it/v1/endpoints/RetrievePeriodicObligationsControllerISpec.scala

```diff
@@ -0,0 +1,191 @@
+import v1.models.errors.{BusinessIdFormatError, FromDateFormatError, MissingTypeOfBusinessError, MtdError, NinoFormatError, StatusFormatError, ToDateFormatError, TypeOfBusinessFormatError}
+import v1.stubs.{AuditStub, AuthStub, DesStub, MtdIdLookupStub}
+
+class RetrievePeriodicObligationsControllerISpec extends IntegrationBaseSpec {
+
+  private trait Test {
+
+    val nino = "AA123456A"
+    val typeOfBusiness = "self-employment"
+    val incomeSourceId = "XAIS123456789012"
+    val fromDate = "2019-01-01"
+    val toDate = "2019-06-06"
+    val status = "Open"
+    val correlationId = "X-123"
+
+    def setupStubs(): StubMapping
+
+    def uri: String = s"/$nino/income-and-expenditure"
+
+    def desUri: String = s"/enterprise/obligation-data/nino/$nino/ITSA"
+
+    def queryParams: Map[String, String] = Map (
+      "from" -> "2019-01-01",
+      "to" -> "2019-06-06"
+    )
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+
+    val responseBody = Json.parse(
+      """{
+        |  "obligations": [
+        |     {
+        |       "typeOfBusiness": "self-employment",
+        |       "businessId": "XAIS123456789012",
+        |       "obligationDetails": [
+        |         {
+        |           "periodStartDate": "2019-01-01",
+        |           "periodEndDate": "2019-06-06",
+        |           "dueDate": "2019-04-30",
+        |           "receivedDate": "2019-04-25",
+        |           "status": "Open"
+        |         }
+        |       ]
+        |    }
+        |  ]
+        |}
+        |""".stripMargin)
+
+    val desResponse: JsValue = Json.parse(
+      """
+        |{
+        |    "obligations": [
+        |        {
+        |            "identification": {
+        |                "incomeSourceType": "ITSB",
+        |                "referenceNumber": "XAIS123456789012",
+        |                "referenceType": "IncomeSourceId"
+        |            },
+        |            "obligationDetails": [
+        |                {
+        |                    "status": "O",
+        |                    "inboundCorrespondenceFromDate": "2019-01-01",
+        |                    "inboundCorrespondenceToDate": "2019-06-06",
+        |                    "inboundCorrespondenceDateReceived": "2019-04-25",
+        |                    "inboundCorrespondenceDueDate": "2019-04-30",
+        |                    "periodKey": "#001"
+        |                }
+        |            ]
+        |        }
+        |    ]
+        |}
+        |""".stripMargin
+    )
+
+    def errorBody(code: String): String =
+      s"""
+         |{
+         |     "code": "$code",
+         |     "reason": "des message"
+         |}
+    """.stripMargin
+
+  }
+
+  "Calling the retrieve periodic obligations endpoint" should {
+
+    "return a 200 status code" when {
+
+      "any valid request is made" in new Test {
```

Comment:
You currently only have a scenario for "DES returns one object with one obligationDetail". Can you also add scenarios for:
- DES returns one object with multiple obligationDetails
- DES returns multiple objects, each with one obligationDetail in them
- DES returns multiple objects, each with multiple obligationDetails in them

Just to verify that our janky reads are working properly for this endpoint

Reply:
^ this again